go-controller/pkg/level_driven_controller/level_driven_controller.go

```diff
@@ -0,0 +1,268 @@
+package level_driven_controller
```

Comment:
yeah I know about that nice lib, but I faced 2 main problems with it:
1. it uses manager to create multiple controller, and a manager starts its own watchers. But for perf reasons, we want to reuse our watch factory, so as I understand it, we need to switch all our controllers to use the same manager to have the minimal amount of watchers.
2. there is no initialSync functionality (again, maybe there is a workaround for that case, but I didn't figure it out), which we use pretty often on startup.
My main goal was not to have a "good level-driven controller", but mostly to generalize our existing code for level-driven controllers (since I had to use it twice in this PR) that we use in ovn-k today. If I get to choose, I would probably prefer switching to controller-runtime instead of improving this controller to be of comparable quality.

I don't think it can be inside util, because this package imports `factory`, and factory import `util`.

Reply:
`util/controller` would be it's own package, not imported by factory.

but probably we could think of a new package as `<newpkg>/factory`, `<newpkg/informer>`, `<newpkg>/controller`

Reply:
new package makes sense, but probably will be good for another PR
I will just rename new pkg to controller for now